en/controllers/middleware.rst

```diff
@@ -229,7 +225,9 @@ their own response. We can see both options in our simple middleware::
                     'expire' => '+ 1 year',
                 ]);
             }
-            return $response;
+            
+            // Pass the modified instances to the next Middleware
+            return $next($request, $response);
```

Comment:
I just spent 15 minutes today trying to figure out why I was getting a 'You returned no response' error on my application, to realise I'd missed this sneeky reassignment of `$response`. This just reads more clearly to me, and is consistant with the core.

* https://github.com/cakephp/cakephp/blob/838d2bb4395b340b6031677109671025042d2812/src/Routing/Middleware/RoutingMiddleware.php#L62
* https://github.com/cakephp/cakephp/blob/838d2bb4395b340b6031677109671025042d2812/src/Error/Middleware/ErrorHandlerMiddleware.php#L81
* https://github.com/cakephp/cakephp/blob/838d2bb4395b340b6031677109671025042d2812/src/Routing/Middleware/AssetMiddleware.php#L88

So if anything, the documentation could be made clearer to explain what this is doing and why. 
The thing which threw me so much was that the variable was being overwritten but still used.